flocker/provision/_install.py

```diff
@@ -43,6 +43,103 @@
 }
 
 
+def task_client_smoke_test():
+    """
+    Check that the CLI is working.
+    """
+    return run_from_args(['flocker-deploy', '--version'])
+
+
+def task_install_cli(distribution, package_source=PackageSource()):
+    """
+    Install flocker CLI on a distribution.
+
+    :param bytes distribution: The distribution the node is running.
+    :param PackageSource package_source: The source from which to install the
+        package.
+    """
+    if package_source.branch:
+        result_path = posixpath.join(
+            '/results/omnibus/', package_source.branch, distribution)
+        base_url = urljoin(package_source.build_server, result_path)
+    else:
+        base_url = None
+    if distribution == 'ubuntu-14.04':
+        commands = [
+            # Ensure add-apt-repository command and HTTPS URLs are supported
+            sudo_from_args([
+                "apt-get", "-y", "install", "apt-transport-https",
+                "software-properties-common"]),
+            # FLOC-1828 TODO - use ARCHIVE_BUCKET rather than clusterhq-archive-testing  # noqa
+            # This needs a release to be done which adds .deb files to clusterhq-archive  # noqa
+            sudo_from_args([
+                'add-apt-repository', '-y',
+                'deb https://clusterhq-archive-testing.s3.amazonaws.com/ubuntu/14.04/$(ARCH) /'  # noqa
+                ])
+            ]
+
+        if base_url:
```

Comment:
Is it the case that the latest package is downloaded and that may be on Buildbot and may be on S3?